### PR TITLE
fixed issue #4450

### PR DESCRIPTION
**Pull request by hyi**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by hyi** [comment]
@pkdash This issue is very important to get fixed earlier rather than later since more problematic resources with versioning chain rule violations could potentially result before the fix gets deployed. It should be a quick review.

File: hs_core/tasks.py [hunk 1]
```diff
@@ -412,6 +412,12 @@ def copy_resource_task(ori_res_id, new_res_id=None, request_username=None):
             new_res = utils.get_resource_by_shortkey(new_res_id)
         utils.copy_and_create_metadata(ori_res, new_res)
 
+        if new_res.metadata.relations.all().filter(type='isVersionOf').exists():
+            # the resource to be copied is a versioned resource, need to delete this isVersionOf
+            # relation element to maintain the single versioning obsolescence chain
+            eid = new_res.metadata.relations.all().filter(type='isVersionOf').first().id
```
Comment (pkdash) [inline hs_core/tasks.py:418]: You can delete like this: `new_res.metadata.relations.all().filter(type='isVersionOf').first().delete()`

**Review by pkdash — CHANGES_REQUESTED** [review]
One minor change for your consideration.

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW BUGS | 0 |
| NEW DUPLICATED LINES DENSITY | 0.0 |
| NEW SECURITY HOTSPOTS | 0 |
| NEW CODE SMELLS | 0 |
| NEW VULNERABILITIES | 0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4453](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4453)

**Comment by hyi** [comment]
@pkdash made changes as you suggested. Thanks

**Review by pkdash — APPROVED** [review]

**Pull request merged**